### PR TITLE
Fix for floating widget when drivers are nudged silent.

### PR DESCRIPTION
**Pull request by prakharritik**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Modify the updateDriverStatus function by adding a flag. On the basis of the flag the widget service will be started. The function has been moved in rideRequestUtils and called when update storage message is sent. 

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestUtils.java [hunk 1]
```diff
@@ -40,15 +42,21 @@
...
 import javax.net.ssl.HttpsURLConnection;
 
+import in.juspay.mobility.BuildConfig;
 import in.juspay.mobility.MainActivity;
 import in.juspay.mobility.R;
 
 public class RideRequestUtils {
     private final int rideReqNotificationId = 5032023;
+
+    private static final String LOG_TAG = "NotificationServices";
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestUtils.java:52]: Rename This Tag

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestUtils.java [hunk 2]
```diff
@@ -264,4 +269,74 @@ public void openApplication(Context context){
         }catch (Exception ignored){}
     }
 
+    private static void updateStorage(String key, String value, Context context){
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestUtils.java:272]: remove this function Just use `sharedPref.edit().putString(storage_key, storage_value).apply();`

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestUtils.java [hunk 3]
```diff
@@ -264,4 +269,74 @@ public void openApplication(Context context){
...
+                connection.setRequestProperty("x-device", deviceDetails);
+                connection.setDoOutput(true);
+                connection.connect();
+
+                // validating the response code
+                int respCode = connection.getResponseCode();
+                InputStreamReader respReader;
+                Log.d(LOG_TAG, "respCode "+ respCode);
+
+                if(respCode == 200 && startWidget == true){
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestUtils.java:313]: no need to create a new if else case it's already there
Just move this to line number 324

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestUtils.java [hunk 4]
```diff
@@ -264,4 +269,74 @@ public void openApplication(Context context){
...
+                    respReader = new InputStreamReader(connection.getInputStream());
+                    Log.d(LOG_TAG, "in 200 "+ respReader);
+                }
+
+                BufferedReader in = new BufferedReader(respReader);
+                String inputLine;
+                while ((inputLine = in.readLine()) != null) {
+                    result.append(inputLine);
+                }
+                updateStorage("DRIVER_STATUS","__failed", context);
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestUtils.java:333]: no need to call this function Just use sharedPref.edit().putString("DRIVER_STATUS","__failed").apply();

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

**Review by Rohit4997 — APPROVED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**